Katello/katello
# Fixes #33449 - Alternate Content Source refreshing support

#### What are the changes introduced in this pull request?
Adds support for refreshing alternate content sources.

Also adds a rake task to refresh all alternate content sources.  This is to be used by the future cron job.

#### Considerations taken when implementing this change?

Alternate content sources are refreshed in two cases: when a user requests it, and when the cron job (to be done) calls the rake task to refresh all ACSs.

We don't want ACS refresh failures to pause a task, since that would block other ACSs from refreshing. So, we just skip the bad ones.

#### What are the testing steps for this pull request?
1) Try refreshing yum and file ACSs and then syncing repositories that point to the same content

To really test refresh: 
- Host a Katello test repo locally and then create an ACS that points to it.
- Create an immediate upstream repo with the same RPMs (like https://jlsherrill.fedorapeople.org/fake-repos/needed-errata/).
- Refresh the ACS and sync the repository (with an immediate download policy).  The sync should download RPMs from your local server since the ACS will be preferred.  You can see this in your local web server logs.

2) Try bulk refreshing ACSs
3) Try the ACS rake task
4) Try bulk destroying ACSs (something unrelated I added since I was already making bulk actions)
